source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACORenderContext.mm

```diff
@@ -116,10 +148,14 @@ - (void)pushBaseCardElementContext:(ACOBaseCardElement *)acoElement
         }
     }
 
+    NSNumber *iconPlacement = nil;
+    if (acoElement.type == ACRActionSet) {
+        iconPlacement = [self.hostConfig getIconPlacement:key];
+    }
+
     std::shared_ptr<BaseElement> baseElement = std::dynamic_pointer_cast<BaseElement>(element);
-    NSNumber *key = [NSNumber numberWithLong:(baseElement->GetInternalId()).Hash()];
 
-    [self addToANewContext:crtl key:key hasSelectAction:hasSelectAction];
+    [self addToANewContext:crtl key:key hasSelectAction:hasSelectAction iconPlacement:iconPlacement];
```

Comment:
Does this misbehave is `iconPlacement` stays `nil`?

Reply:
No, it would not. The function will check the nullability and wouldn't add it for tracking.